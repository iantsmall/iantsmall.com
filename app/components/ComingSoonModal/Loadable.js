/**
 *
 * Asynchronously loads the component for ComingSoonModal
 *
 */

import loadable from 'utils/loadable';

export default loadable(() => import('./index'));
