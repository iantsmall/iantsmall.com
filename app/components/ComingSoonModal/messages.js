/*
 * ComingSoonModal Messages
 *
 * This contains all the text for the ComingSoonModal component.
 */

import React from 'react';

import { defineMessages } from 'react-intl';

export const scope = 'app.components.ComingSoonModal';

const features = [
  'Welcoming Page',
  'About Me Page',
  'Development Portfolio',
  'Contact Information',
];
const bugs = [];

export default defineMessages({
  title: {
    id: `${scope}.title`,
    defaultMessage: 'Coming Soon',
  },
  body: {
    id: `${scope}.body`,
    defaultMessage: '{summary} {features} {bugs}',
    values: {
      summary: <p>This is a simple road map for coming features.</p>,
      features: (
        <div id="planned-features">
          <h2>Planned Features</h2>
          <ul>
            {features.map(feature => (
              <li key={`feature_${feature}`}>{feature}</li>
            ))}
          </ul>
        </div>
      ),
      bugs: (
        <div id="bug-fixes">
          <h2>Known Bugs</h2>
          <ul>
            {bugs.map(bug => (
              <li key={`bug_${bug}`}>{bug}</li>
            ))}
          </ul>
        </div>
      ),
    },
  },
});
