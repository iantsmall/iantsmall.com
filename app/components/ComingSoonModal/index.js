/**
 *
 * ComingSoonModal
 *
 */

import React, { memo } from 'react';
import PropTypes from 'prop-types';
// import styled from 'styled-components';
import { Modal } from 'react-bootstrap';

import { FormattedMessage } from 'react-intl';
import messages from './messages';

function ComingSoonModal({ show, onHide }) {
  return (
    <Modal id="coming-soon" show={show} onHide={onHide}>
      <Modal.Header closeButton>
        <Modal.Title>
          <FormattedMessage {...messages.title} />
        </Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <FormattedMessage {...messages.body} />
      </Modal.Body>
    </Modal>
  );
}

ComingSoonModal.propTypes = {
  show: PropTypes.bool,
  onHide: PropTypes.func,
};

export default memo(ComingSoonModal);
