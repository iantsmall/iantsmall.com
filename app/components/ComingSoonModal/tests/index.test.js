/**
 *
 * Tests for ComingSoonModal
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
// import { IntlProvider } from 'react-intl';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import render from 'utils/testRenderWithContext';
import ComingSoonModal from '../index';
// import { DEFAULT_LOCALE } from '../../../i18n';

describe('<ComingSoonModal />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<ComingSoonModal />);
    expect(spy).not.toHaveBeenCalled();
  });

  it.skip('Expect to have additional unit tests specified', () => {
    expect(true).toEqual(false);
  });

  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<ComingSoonModal />);
    expect(firstChild).toMatchSnapshot();
  });
});
