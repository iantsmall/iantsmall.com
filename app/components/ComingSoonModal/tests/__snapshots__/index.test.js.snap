// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComingSoonModal /> Should render and match the snapshot 1`] = `null`;
